projects/lights_control/inc/lights_gpio.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include "event_queue.h"
+
+#include "status.h"
+
+typedef enum { LIGHTS_BOARD_FRONT, LIGHTS_BOARD_REAR, NUM_LIGHTS_BOARDS } LightsBoard;
+
+// used for figuring out which board we're operating on is called before lights_gpio_init since
+// lights_gpio_init needs to know which board it needs to initialize
+StatusCode lights_gpio_get_lights_board(LightsBoard *board);
+
+// initializes all the GPIO peripherals
+StatusCode lights_gpio_init();
+
+// sets the state of every peripheral related to a specific event
+StatusCode lights_gpio_set(Event *e);
```

Comment:
`const Event *e`? Also you may want to add event generators for on and off since your `e->data` is just a number but you are interpreting it as a state?

Reply:
Also, `set` implies you're providing a state. This is more of processing an event.

Reply:
changed it to `lights_gpio_process_event(const Event *)`